presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergEqualityDeleteAsJoin.java

```diff
@@ -383,24 +383,31 @@ private ImmutableMap<VariableReferenceExpression, ColumnHandle> createAssignment
                             if (!partitionField.transform().isIdentity()) {
                                 Type partitionFieldType = partitionField.transform().getResultType(sourceField.type());
                                 VariableReferenceExpression variableReference = variableAllocator.newVariable(partitionField.name(), toPrestoType(partitionFieldType, typeManager));
-                                IcebergColumnHandle columnHandle = new IcebergColumnHandle(
-                                        ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
-                                        toPrestoType(partitionFieldType, typeManager),
-                                        Optional.empty(),
-                                        PARTITION_KEY);
-                                unselectedAssignmentsBuilder.put(variableReference, columnHandle);
+                                if (!selectedFields.contains(partitionField.fieldId())) {
+                                    IcebergColumnHandle columnHandle = new IcebergColumnHandle(
+                                            ColumnIdentity.createColumnIdentity(partitionField.name(), partitionField.fieldId(), partitionFieldType),
+                                            toPrestoType(partitionFieldType, typeManager),
+                                            Optional.empty(),
+                                            PARTITION_KEY);
+                                    unselectedAssignmentsBuilder.put(variableReference, columnHandle);
+                                    selectedFields.add(partitionField.fieldId());
```

Comment:
This is to ensure no duplicates are added? `unselectedFields` is already a set so it shouldn't happen.